lib/parser/provenance.cc

```diff
@@ -241,6 +253,26 @@ const SourceFile *AllSources::GetSourceFile(
       origin.u);
 }
 
+std::optional<std::pair<int, int>> AllSources::GetSourcePosition(
+    Provenance prov) const {
+  const Origin &origin{MapToOrigin(prov)};
+  if (auto inc = std::get_if<Inclusion>(&origin.u)) {
+    std::size_t offset{origin.covers.MemberOffset(prov)};
+    return {std::pair<int, int>(inc->source.FindOffsetLineAndColumn(offset))};
+  } else {
+    return std::nullopt;
+  }
+}
+
+std::optional<ProvenanceRange> AllSources::GetFirstFileProvenance() {
+  for (unsigned int i = 0; i < origin_.size(); i++) {
+    if (std::get_if<Inclusion>(&origin_[i].u)) {
```

Comment:
Since you don't use the result of `get_if`, `std::holds_alternative<Inclusion>(origin_[i].u)` suffices.